x/relaystate.go

```diff
@@ -0,0 +1,47 @@
+package x
+
+import (
+	"net/http"
+
+	"github.com/gorilla/sessions"
+	"github.com/pkg/errors"
+)
+
+// SessionGetRelayState returns a string of the content of the relaystate for the current session.
+func SessionGetStringRelayState(r *http.Request, s sessions.StoreExact, id string, key interface{}) (string, error) {
+
+	cipherRelayState := r.PostForm.Get("RelayState")
```

Comment:
This should be underscored and lowercase:

	cipherRelayState := r.PostForm.Get("relay_state")

Reply:
We have to use this "RelayState" notation because it is a SAML standard